app/shared-components/PromoModal/PromoModal.jsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * Base component for all of the Promo Modals
+ *
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2019 Ghostery, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClassNames from 'classnames';
+import Modal from '../Modal';
+import InsightsPromoModalContent from '../ModalContent/InsightsPromoModalContent';
+import PlusPromoModalContent from '../ModalContent/PlusPromoModalContent';
+import PremiumPromoModalContent from '../ModalContent/PremiumPromoModalContent';
+import history from '../../panel/utils/history';
+import { sendMessage } from '../../panel/utils/msg';
+import globals from '../../../src/classes/Globals';
+import ModalExitButton from '../../panel/components/BuildingBlocks/ModalExitButton';
+
+const DOMAIN = globals.DEBUG ? 'ghosterystage' : 'ghostery';
+const INSIGHTS = 'insights';
+const PLUS = 'plus';
+const PREMIUM = 'premium';
+
+/**
+ * A base functional component for Promo Modals
```

Comment:
Update description to reflect that this is a class component, not a functional one

Reply:
Updated comment to a class component